server/sonar-server/src/main/java/org/sonar/server/component/ws/ProjectMeasuresQueryFactory.java

```diff
@@ -122,4 +142,16 @@ private static double parseValue(String value) {
     }
   }
 
+  private static void checkValue(Criterion criterion) {
+    checkArgument(criterion.getValue() != null, "Value cannot be null for '%s'", criterion.getKey());
+  }
+
+  private static void checkOperator(Criterion criterion) {
+    checkArgument(criterion.getOperator() != null, "Operator cannot be null for '%s'", criterion.getKey());
+  }
+
+  @FunctionalInterface
+  private interface CriterionProcessor {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Drop this interface in favor of "java.util.function.BiConsumer<FilterParser$Criterion,ProjectMeasuresQuery>". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1711)